hyperspy/events.py

```diff
@@ -90,22 +187,28 @@ def connect(self, function, nargs='all'):
         self._connected[nargs].add(function)
 
     def disconnect(self, function):
+        """
+        Disconnects a function from the event. The passed function will be
+        disconnected irregardless of which 'nargs' argument was passed to
+        connect().
+        """
         for c in self._connected.itervalues():
             if function in c:
                 c.remove(function)
 
-    def trigger(self, *args, **kwargs):
-        if not self.suppress:
-            for nargs, c in self._connected.iteritems():
+    def trigger(self, *args):
+        if not self._suppress:
+            # Loop on copy to deal with callbacks which change connections
+            for nargs, c in self._connected.copy().iteritems():
                 if nargs is 'all':
                     for f in c:
-                        f(*args, **kwargs)
```

Comment:
Sure. I'll undo it. Do you know if inspect can figure out which kwargs a
function accepts? If so, we could avoid som "Key word argument not
accepted" errors. Or do we want to leave that to the users (just add a
**kwargs to deal with unwanted keyword args)?